dhall/src/Dhall/Import.hs

```diff
@@ -494,7 +495,10 @@ loadImportWithSemanticCache
 loadImportWithSemanticCache
   import_@(Chained (Import (ImportHashed (Just semanticHash) _) _)) = do
     Status { .. } <- State.get
-    mCached <- liftIO $ fetchFromSemanticCache semanticHash
+    mCached <-
+        case _semanticCacheMode of
+            UseSemanticCache -> liftIO $ fetchFromSemanticCache semanticHash
+            IgnoreSemanticCache -> pure Nothing
 
     case mCached of
         Just bytesStrict -> do
```

Comment:
[A bit further below](https://github.com/dhall-lang/dhall-haskell/pull/1290/files#diff-c4df37703a934d22d0c320bad5bf9143R528) we also `writeToSemanticCache` – IMHO it's not necessary to disable this, but we could.